client/my-sites/upgrades/domain-management/name-servers/dns-templates.jsx

```diff
@@ -2,23 +2,52 @@
  * External dependencies
  */
 import React, { Component } from 'react';
-import { find } from 'lodash';
+import { find, replace } from 'lodash';
 import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
+import ButtonGroup from 'components/button-group';
 import Card from 'components/card';
+import { dnsTemplates } from 'lib/domains/constants';
 import DnsTemplateButton from './dns-template-button';
-import Office365 from '../dns/office-365';
+import EmailProvider from '../dns/email-provider';
 
 class DnsTemplates extends Component {
 	constructor( props ) {
 		super( props );
+		const { translate } = this.props;
 		this.state = {
 			currentComponentName: null,
+			currentProviderCardName: null,
 			templates: [
-				{ name: 'Office 365', Component: Office365 }
+				{
+					name: 'G Suite',
+					label: translate( 'G Suite Verification Token - from the TXT record verification' ),
+					placeholder: 'google-site-verification=...',
+					validationPattern: /^google-site-verification=\w{43}$/,
+					dnsTemplate: dnsTemplates.G_SUITE
+				},
+				{
+					name: 'Office 365',
+					label: translate( 'Office 365 Verification Token - from the TXT record verification' ),
+					placeholder: 'MS=ms...',
+					validationPattern: /^MS=ms\d{4,20}$/,
```

Comment:
Every documentation and examples I can find suggest that the digits part is always 8 characters long - do you happen to know if that's true?